tests/cases/03_curd.ts

```diff
@@ -578,3 +578,51 @@ testWithTestDBClient("testFindEmptyAsyncIteration", async (db) => {
 
   await db.collection("mongo_test_users").drop();
 });
+
+testWithTestDBClient("testFindWithMaxTimeMS", async (db) => {
+  const users = db.collection<IUser>("mongo_test_users");
+  for (let i = 0; i < 10; i++) {
+    await users.insertOne({
+      username: "testFindWithMaxTimeMS",
+      password: "pass1",
+      uid: i,
+    });
+  }
+  const users1 = await users.find({
+    uid: 0,
+  }, { maxTimeMS: 100 }).toArray();
+
+  assertEquals(users1.length, 1);
+
+  const user1 = await users.findOne({
+    uid: 0,
+  }, { maxTimeMS: 100 });
+
+  assertEquals(user1!.uid, 0);
+
+  try {
+    await users.find({
+      uid: 0,
+      $where: "sleep(5) || true",
+    }, { maxTimeMS: 1 }).toArray();
+    assert(false);
+  } catch (e) {
+    assertEquals(Boolean(e.ok), false);
+    assertEquals(e.codeName, "MaxTimeMSExpired");
+    assertEquals(e.errmsg, "operation exceeded time limit");
```

Comment:
This is failing on 4.0 but I don't know what the error message actually is, there. Do you have any insight on how to differently check/assert depending on the mongo versions?

Reply:
im a bit again doing assert based on mongodb version as they should be compatible in that case